### DISCUSSION TITLE
Does the right-click function come back when I press the Galaxy Tab s pen button?

### DISCUSSION THREAD

**Discussion by shootmg50** [LizardByte/community#354] [Sunshine] [answered]
When I use Moonlight in GeForce Shield, I right-clicked the Galaxy Tab s pen button, but I can't right-click the s pen button in Sunshine Moonlight. Can the spen button function come back in the future?

**Comment by ReenigneArcher** [comment]
> spen

I don't even know what this is...

**Reply by shootmg50** [reply]
sorry galaxy tab s pen.

**Reply by Seeevers** [reply]
I've been trying to figure this out for days.

**Comment by cgutman** [comment, marked as answer]
Since Sunshine v0.21.0, pen and touch events are natively passed to applications on the host. The stylus button is passed to applications on the host as the barrel button (`PEN_FLAG_BARREL`) as documented here: https://learn.microsoft.com/en-us/windows/win32/inputmsg/pen-flags-constants

In Windows, the barrel button should be configured to act as right click by default but you can confirm this in the Pen and Touch Control Panel applet.

![image](https://github.com/LizardByte/Sunshine/assets/2695644/16dbc780-0a0d-4914-9b43-db8c4f76d848)

The way it works is if you tap the stylus on the screen while holding the stylus button, it should act as a right click. You can also tap and hold to right click (same with touch), though it can be tricky to keep the stylus stationary on a small screen to perform that gesture.

In the latest [nightly builds](https://github.com/LizardByte/Sunshine/releases/nightly-dev), you can also disable native pen/touch support on the input tab of the web UI and that will make Sunshine behave like GFE did for pen and touch input (emulating a mouse).

**Reply by shootmg50** [reply]
I've stopped by again to look for solutions. Today, I remembered and searched, and I'm very happy to have found a solution. Thank you. I appreciate you creating such a great program.

**Reply by Bara3d** [reply]
one problem I've faced with going to the trackpad emulation option (which so far is the best option for me) is if I want to shift+right click, it doesnt register the shift. Im wondering if this is a bug, or if theres a work around for it

**Reply by ForteDexe** [reply]
Although the pressure is regconize, but this window setting is no longer exist in windows 11, is there anyway to work around ?
